lua/starfall/libs_sv/constraint.lua

```diff
@@ -0,0 +1,107 @@
+
+-------------------------------------------------------------------------------
+-- Constraint functions
+-------------------------------------------------------------------------------
+
+assert( SF.GetTypeDef( "Entity" ) )
+
+--- Constraint library. Used for various entity constraints
+-- @name constraint
+-- @server
+-- @class library
+-- @libtbl constraint_lib
+local constraint_lib, _ = SF.Libraries.Register( "constraint" )
+
+local ents_metatable = SF.GetTypeDef( "Entity" )
+local wrap, unwrap = ents_metatable.__wrap, ents_metatable.__unwrap
+local vwrap, vunwrap = SF.WrapObject, SF.UnwrapObject
+
+do
+	local P = SF.Permissions
+	P.registerPrivilege( "constraint.any", "Constraint", "Allows the user to create constraints" )
+end
+
+--- Creates a table of entities recursively constrained to an entity
+-- @param ent Base entity of search
+-- @param default Default table to return
+-- @return Table of constrained entities
+function constraint_lib.getAllConstrainedEntities( ent, default )
+	SF.CheckType( ent, ents_metatable )
+
+	if not SF.Permissions.check( SF.instance.player, ent, "constraint.any" ) then SF.throw( "Insufficient permissions", 2 ) end
+
+	ent = unwrap( ent )
+
+	if default ~= nil then
+		SF.CheckType( default, "table" )
+	else
+		default = {}
+	end
+
+	local valid = SF.Entities.IsValid
+	if not valid( ent ) then return {} end
+
+	local constrainedEnts = constraint.GetAllConstrainedEntities( ent, {} )
+
+	local entsTable = {}
+	for _, Entity in pairs( constrainedEnts ) do
+		table.insert( entsTable, wrap( Entity ) )
+	end
+
+	if #entsTable == 0 then
+		entsTable = default
+	end
+
+	return entsTable
+end
+
+--- Creates a table of constraints on an entity
+-- @param ent Target entity
+-- @return Table of entity constraints
+function constraint_lib.getTable( ent )
+	SF.CheckType( ent, ents_metatable )
+
+	if not SF.Permissions.check( SF.instance.player, ent, "constraint.any" ) then SF.throw( "Insufficient permissions", 2 ) end
+
+	ent = unwrap( ent )
+
+	local valid = SF.Entities.IsValid
+	if not valid( ent ) then return {} end
+
+	local constraints = constraint.GetTable( ent )
+
+	local ret = {}
+	for k, v in pairs( constraints ) do
```

Comment:
Most of this can be done using SF.Santitize ( check spelling and caps ). That will step through, even recursively, and wrap things that need to be wrapped. You'd only need to set Type then. That's the safest way incase anything is changed in how the constraints table is formatted in the future.